src/include/distributed/multi_client_executor.h

```diff
@@ -72,6 +75,28 @@ typedef enum
 } BatchQueryStatus;
 
 
+/* Enumeration to track what the task associated with a connection is waiting for */
+typedef enum WaitStatus
```

Comment:
The names of these enums are really tripping me up. They're not always _track[ing] what the task associated with a connection is waiting for_, right? For instance, the error state is just a signal that an error has occurred, not e.g. that we're waiting for an error.